torch/_inductor/triton_heuristics.py

```diff
@@ -1191,9 +1191,10 @@ def foreach(meta, num_warps, filename=None):
 
 def grid(*numels):
     """Helper function to compute triton grids"""
-
     if len(numels) == 1 and isinstance(numels[0], tuple):
-        numels = numels[0]
+        # when passed as a single tuple, calling convention order is not
+        # followed, so we need to reverse to match calling conversion order
+        numels = numels[0][::-1]
```

Comment:
I don't understand why this change is needed, aren't we not using this function for triton kernels?

Reply:
cached_autotune uses the launder function which requires everything to be 3tuple since it unpacks.

https://github.com/pytorch/pytorch/blob/c14c4efc0eb97ccc471ac6cbdcc659417227ae97/torch/_inductor/triton_heuristics.py#L316-L319

I was using this function to convert 1tuple or 2tuple into 3 tuple. I can either write my own function to do this or is there a more pythonic way to do this converstion?

Reply:
Since we are generating the grid function, can't we just make it return a 3-tuple?